src/Uno.UI/UI/Xaml/Media/Imaging/RenderTargetBitmap.skia.cs

```diff
@@ -1,52 +1,91 @@
 ﻿#nullable enable
+using System;
 using System.Runtime.InteropServices;
 using Windows.Foundation;
+using Windows.UI.Composition;
+using Uno.UI.Xaml.Media;
+using SkiaSharp;
+
 namespace Windows.UI.Xaml.Media.Imaging
 {
 	partial class RenderTargetBitmap
 	{
+		private const int _bitsPerPixel = 32;
+		private const int _bitsPerComponent = 8;
+		private const int _bytesPerPixel = _bitsPerPixel / _bitsPerComponent;
+
 		delegate void SwapColor(ref byte[] buffer, int byteCount);
 
-		static readonly SwapColor? PlatformSwap = RuntimeInformation.IsOSPlatform(OSPlatform.OSX)
-			? new SwapColor(SwapRB)
-			: default(SwapColor);
+		private static readonly SwapColor? _platformSwap = RuntimeInformation.IsOSPlatform(OSPlatform.OSX) ? SwapRB : default;
+
+		private static ImageData Open(byte[] buffer, int bufferLength, int width, int height)
+		{
+			if (_platformSwap is not null)
+			{
+				var swappedBuffer = default(byte[]);
+				EnsureBuffer(ref swappedBuffer, bufferLength);
+				Array.Copy(buffer, swappedBuffer!, bufferLength);
+				_platformSwap(ref swappedBuffer!, bufferLength);
+				buffer = swappedBuffer;
+			}
+
+			var bufferHandle = GCHandle.Alloc(buffer, GCHandleType.Pinned);
+			try
+			{
+				// Note: We use the FromPixelCopy which will create a clone of the buffer, so we are ready to be re-used to render another UIElement.
+				// (It's needed also for if we swapped the buffer since we are not maintaining a ref on the swappedBuffer)
```

Comment:
```suggestion
				// Note: We use the FromPixelCopy which will create a clone of the buffer, so we are ready to be re-used to render another UIElement.
				// (It's needed also if we swapped the buffer since we are not maintaining a ref on the swappedBuffer)
```